homeassistant/components/blue_current/__init__.py

```diff
@@ -0,0 +1,178 @@
+"""The Blue Current integration."""
+from __future__ import annotations
+
+from contextlib import suppress
+from datetime import datetime
+from typing import Any
+
+from bluecurrent_api import Client
+from bluecurrent_api.exceptions import (
+    BlueCurrentException,
+    InvalidApiToken,
+    RequestLimitReached,
+    WebsocketError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_NAME, CONF_API_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_call_later
+
+from .const import DOMAIN, EVSE_ID, LOGGER, MODEL_TYPE
+
+PLATFORMS = [Platform.SENSOR]
+CHARGE_POINTS = "CHARGE_POINTS"
+DATA = "data"
+SMALL_DELAY = 1
+LARGE_DELAY = 20
+
+GRID = "GRID"
+OBJECT = "object"
+VALUE_TYPES = ["CH_STATUS"]
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up Blue Current as a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    client = Client()
+    api_token = config_entry.data[CONF_API_TOKEN]
+    connector = Connector(hass, config_entry, client)
+
+    try:
+        await connector.connect(api_token)
+    except InvalidApiToken:
+        LOGGER.error("Invalid Api token")
+        return False
+    except BlueCurrentException as err:
+        raise ConfigEntryNotReady from err
+
+    hass.async_create_task(connector.start_loop())
```

Comment:
Use `hass.async_create_background_task` instead. Otherwise this task will hold up startup and eventually create a warning when startup times out.